packages/babel-plugin-transform-class-properties/test/fixtures/regression/T7364/expected.js

```diff
@@ -2,8 +2,18 @@ class MyClass {
   constructor() {
     var _this = this;
 
-    this.myAsyncMethod = babelHelpers.asyncToGenerator(function* () {
-      console.log(_this);
+    Object.defineProperty(this, "myAsyncMethod", {
+      enumerable: true,
+      writable: true,
+      value: (() => {
+        var _ref = babelHelpers.asyncToGenerator(function* () {
+          console.log(_this);
+        });
+
+        return function value() {
```

Comment:
```
return function value() {
                ^^^^^
```
Kinda meaningless function name inference here. The surrounding IIFE's sole purpose is to rename the function returned from `asyncToGenerator`.